sources/api/pluto/src/main.rs

```diff
@@ -146,23 +125,22 @@ use error::PlutoError;
 
 type Result<T> = std::result::Result<T, PlutoError>;
 
-async fn get_text_from_imds(client: &Client, uri: &str, session_token: &str) -> Result<String> {
-    client
-        .get(uri)
-        .header("X-aws-ec2-metadata-token", session_token)
-        .send()
-        .await
-        .context(error::ImdsRequest { method: "GET", uri })?
-        .error_for_status()
-        .context(error::ImdsResponse { uri })?
-        .text()
+async fn get_max_pods(imdsclient: &mut ImdsClient) -> Result<String> {
```

Comment:
Nit: It's typically good Rust style to split words in variables with underscore: `imds_client`